public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -270,27 +271,26 @@ export class CloudWatchDatasource
     metricQueries: CloudWatchMetricsQuery[],
     options: DataQueryRequest<CloudWatchQuery>
   ): Observable<DataQueryResponse> => {
-    const validMetricsQueries = metricQueries
-      .filter(this.filterQuery)
-      .map((item: CloudWatchMetricsQuery): MetricQuery => {
-        item.region = this.templateSrv.replace(this.getActualRegion(item.region), options.scopedVars);
-        item.namespace = this.replace(item.namespace, options.scopedVars, true, 'namespace');
-        item.metricName = this.replace(item.metricName, options.scopedVars, true, 'metric name');
-        item.dimensions = this.convertDimensionFormat(item.dimensions ?? {}, options.scopedVars);
-        item.statistic = this.templateSrv.replace(item.statistic, options.scopedVars);
-        item.period = String(this.getPeriod(item, options)); // use string format for period in graph query, and alerting
-        item.id = this.templateSrv.replace(item.id, options.scopedVars);
-        item.expression = this.templateSrv.replace(item.expression, options.scopedVars);
-        item.sqlExpression = this.templateSrv.replace(item.sqlExpression, options.scopedVars, 'raw');
-
-        return {
-          intervalMs: options.intervalMs,
-          maxDataPoints: options.maxDataPoints,
-          ...item,
-          type: 'timeSeriesQuery',
-          datasource: this.getRef(),
-        };
-      });
+    const validMetricsQueries = metricQueries.filter(this.filterQuery).map((q: CloudWatchMetricsQuery): MetricQuery => {
+      const query = migrateMetricQuery(q);
+      query.region = this.templateSrv.replace(this.getActualRegion(query.region), options.scopedVars);
```

Comment:
it's not clear the difference between `migrateMetricQuery` and the replacements here. Maybe it'd help readability to move these to something like `replaceVars(query)` to make a clear differentiation with the migration above.

Reply:
I agree. I'll add a replaceVars func!